### PR TITLE
added inital testing framework

### PR DESCRIPTION
**Pull request by btkostner**
This adds a very small unit test section, it includes putting the server up and making sure `/` is accessible, and that the whole project lints without errors. It is not a final product of this branch, but it includes linting improvements and `package.json` improvements that would be nice to use.

### CODE REVIEW

File: config.example.js [hunk 1]
```diff
@@ -61,7 +61,6 @@ export const aptly = {
 export const database = 'mongodb://localhost/houston-dev'
 
 export const server = {
-  port: 3000,
```
Comment (fabianthoma) [inline config.example.js]: Why are you removing the port here, this port configuration should stay seperate from the public server url (where you put it) as we want to route through nginx usually.

File: package.json [hunk 2]
```diff
@@ -40,17 +40,32 @@
   "devDependencies": {
     "babel-eslint": "^4.1.8",
     "babel-node-debug": "^2.0.0",
-    "standard": "^6.0.5"
+    "mocha": "^2.4.5",
+    "proxyquire": "^1.7.4",
+    "should": "^8.2.2",
+    "snazzy": "^3.0.0",
+    "standard": "^6.0.5",
+    "supertest": "^1.2.0"
```
Comment (fabianthoma) [inline package.json]: `mocha-standard` is missing

File: config.example.js (hunk 1 above)
Comment (btkostner) [inline config.example.js]: I wasn't aware you used it. It is set-able with a env variable, but I feel this setting is somewhat repetitive, so I'll make it optional.

**Comment by btkostner** [comment]
should wait until the optional profile PR is submitted

**Pull request merged**